integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,239 @@
+
+import java.util.Properties;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.AndTerm;
+import javax.mail.search.BodyTerm;
+import javax.mail.search.SearchTerm;
+import javax.mail.search.SubjectTerm;
+
+/**
+ * Util class for GreenMail Server
+ */
+public class GreenMailServer {
+    protected static Log log = LogFactory.getLog(GreenMailServer.class);
+    private static GreenMail greenMail;
+    private static GreenMailUser greenMailUser;
+    private static String EMAIL_INBOX = "INBOX";
+    private static String EMAIL_TRASH = "Trash";
+    private static int WAIT_TIME_MS = 180 * 1000;
+
+    /**
+     * Start the server and add the user
+     */
+    public void startServer() {
+        ServerSetup popSetup = new ServerSetup(3110, "localhost", "pop3");
+        ServerSetup imapSetup = new ServerSetup(3143, "localhost", "imap");
+        ServerSetup[] setups = new ServerSetup[2];
+        setups[0] = popSetup;
+        setups[1] = imapSetup;
+        greenMail = new GreenMail(setups);
+        greenMail.start();
+        greenMailUser = greenMail.setUser("wso2@localhost", "wso2", "wso2");
+        log.info("GreenMail Server started and user added!");
+    }
+
+    /**
+     * Stop server
+     */
+    public void stopServer() {
+        greenMail.stop();
+        log.info("GreenMail Server stopped!");
+    }
+
+    public static GreenMailUser getGreenMailUser() {
+        return greenMailUser;
+    }
+
+    /**
+     * Connecting to the pop3 mail store
+     *
+     * @return pop3 store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getPopConnection() throws ESBMailTransportIntegrationTestException, MessagingException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName("pop3", "127.0.0.1", 3110, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        Store store = session.getStore(urlName);
+        store.connect();
+        return store;
+    }
+
+    /**
+     * Connecting to the imap mail store
+     *
+     * @return imap store connection
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    private static Store getImapConnection() throws ESBMailTransportIntegrationTestException, MessagingException {
+        Properties props = new Properties();
+        Session session = Session.getInstance(props);
+        URLName urlName = new URLName("imap", "127.0.0.1", 3143, null, greenMailUser.getLogin(),
+                greenMailUser.getPassword());
+        Store store = session.getStore(urlName);
+        store.connect();
+        return store;
+    }
+
+    /**
+     * Check mail folder for an email using subject
+     *
+     * @param emailSubject
```

Comment:
Let's describe the parameters, shall we correct in all places?

Reply:
done